torch/csrc/autograd/record_function.cpp

```diff
@@ -72,7 +121,8 @@ void RecordFunction::processCallbacks() {
 }
 
 RecordFunction::~RecordFunction() {
-  if (has_callbacks) {
+  if (initialized_ &&
```

Comment:
When would a destructor get called with a vaule of `initialized_` unset? A failure of constructor doesn't invoke the destructor, and if I understand correctly you always exit the constructor with `initialized_ == true`

Reply:
initialized_ is set to false if you use default RecordFunction() constructor without parameters